src/main/java/duke/task/Deadline.java

```diff
@@ -0,0 +1,47 @@
+package duke.task;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+
+/**
+ * Represents a Deadline task.
+ */
+public class Deadline extends Task {
+
+    private LocalDateTime deadline;
+
+    /**
+     * Creates a Deadline task with the associated description and deadline.
+     * @param desc Describes the deadline.
+     * @param deadline The deadline that the task is due by.
+     */
+    public Deadline(String desc, String deadline) {
+        super(desc);
+        this.deadline = stringToDateTime(deadline);
+    }
+
+    public Deadline(String desc, boolean isDone, String deadline) {
+        super(desc, isDone);
+        this.deadline = stringToDateTime(deadline);
+    }
+
+    public LocalDateTime getDeadline() {
+        return this.deadline;
+    }
+
+    public void setDeadline(String deadline) {
+        this.deadline = stringToDateTime(deadline);
+    }
+
+    private LocalDateTime stringToDateTime(String deadline) {
```

Comment:
Great use of existing Java libraries to handle date & time formats! I manually coded the string output for every month and day which made my code look very messy... Learnt something new!